### PR TITLE
Convert namedtuples to class syntax

### PR DESCRIPTION
**Pull request by srittau**

### CODE REVIEW

**Comment by srittau** [comment]
I don't know why pytype tests fail in files that have not been changed. Cc @rchen152

File: stdlib/2/os/__init__.pyi [hunk 1]
```diff
@@ -349,20 +356,3 @@ if sys.version_info < (3, 0):
 P_ALL: int
 WEXITED: int
 WNOWAIT: int
-
-if sys.version_info >= (3, 3):
```
Comment (JelleZijlstra) [inline stdlib/2/os/__init__.pyi]: What's going on here?

**Comment by JelleZijlstra** [comment]
As Guido pointed out in the mypy issue, we need to make sure other type checkers support this syntax in stubs first.

File: stdlib/2/os/__init__.pyi (hunk 1 above)
Comment (JelleZijlstra) [reply stdlib/2/os/__init__.pyi]: Oh I see, thanks. I didn't realize this was a Python 2 stub.

**Comment by srittau** [comment]
Missing support could be the reason why pytype fails, although the error messages are a bit weird.

**Comment by rchen152** [comment]
I'll investigate the pytype failures. My first guess would be that pytype expects `class Foo(NamedTuple(...))` rather than `clas Foo(NamedTuple): ...`, hence the error message about encountering a closing parenthesis when another opening one is expected.

**Comment by srittau** [comment]
I have created a "deferred" label to this PR and tagged it with it.

**Comment by rchen152** [comment]
I've changed the pytype parser to support this syntax, but our release process is a little backed up right now - I'll update again when the change is released.

**Comment by srittau** [comment]
I just checked PyCharm, and it is supported. There is also the [Microsoft language server for Visual Studio Code](https://github.com/Microsoft/vscode-python). Cc @mrkmndz @msfterictraut Do pyre and pyright support class-based NamedTuple syntax?

**Comment by mrkmndz** [comment]
Yup that's supported in Pyre

**Comment by srittau** [comment]
VSCode also supports it.

**Comment by erictraut** [comment]
It's supported in pyright also.

**Comment by srittau** [comment]
It seems to be supported by all type checkers. Merging.

**Pull request merged**